DSpace/dspace-angular
# Simplify bitstream drag and drop

## References
This PR closes #679

## Description
This PR simplifies the bitstream drag-and-drop feature to send patch requests as soon as the bitstream is dropped, instead of storing a custom order in the object-updates store.

## Instructions for Reviewers
Changes made:
- Completely removed custom order feature in object-updates reducer and service
- Refactored `AbstractPaginatedDragAndDropListComponent` to send out an Output event containing the fromIndex, toIndex and a callback function when an object is dropped
- Refactored `ItemBitstreamsComponent` to remove "move patch operations" from the `submit()` method and instead create a new method `dropBitstream()`, triggered by the output of the drag-and-drop component. This new method creates a move patch operation and sends it to the REST API. It calls the event's callback function whenever a response returns.
- Modified / Added test cases to match these refactoring changes

How to test:
Go to an item's edit-bitstream page.
- Drag a bitstream to a new location within the same page. The bitstream should instantly move to that location and a patch request is sent automatically
- Drag a bitstream to a new page. The new page should start loading and a patch request is sent automatically. As soon as the move has completed, the new page should be loaded.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

    * Include tests for different user types (if behavior differs), including: (1) Anonymous user, (2) Logged in user (non-admin), and (3) Administrator.
    * Include tests for error scenarios, e.g. when errors/warnings should appear (or buttons should be disabled).
    * For bug fixes, include a test that reproduces the bug and proves it is fixed. For clarity, it may be useful to provide the test in a separate commit from the bug fix.
